File: api/src/main/java/club/yunzhi/smartcommunity/service/PartyBuildingService.java
package club.yunzhi.smartcommunity.service;

import club.yunzhi.smartcommunity.entity.Duty;
import club.yunzhi.smartcommunity.entity.PartyBuilding;

import java.util.List;

/**
 * 党建管理
 */
public interface PartyBuildingService {

  void deleteById(Long id);

  /**
   * 返回所有的党建实体根据当前登陆用户的区域类型
   *
   * @return
   */
  List<PartyBuilding> getAllOfCurrentUserDistrict();

  /**
   * 根据区域id获取所有该区域的党建信息
   *
   * @param districtId
   * @return
   */
  List<PartyBuilding> getAllByDistrictId(Long districtId);

  /**
   * 根据id获取实体
   *
   * @param dutyId     岗位id
   * @param districtId 区域ID
   */
  PartyBuilding getByDutyIdAndDistrictId(Long dutyId, Long districtId);

  /**
   * 查询当前登录用户所在区域中对应的某个职位
   *
   * @param dutyId 职位ID
   */
  PartyBuilding getByDutyIdOfCurrentDistrict(Long dutyId);

  PartyBuilding getById(Long id);

  /**
   * 新增党建管理-职务人员信息
   *
   * @param partyBuilding
   * @return
   */
  PartyBuilding save(PartyBuilding partyBuilding);

  /**
   * 保存
   *
   * @param duty         岗位
   * @param personalName 人员名称
   */
  PartyBuilding saveOfCurrentDistrict(Duty duty, String personalName);

  /**
   * 设置数据
   *
   * @param partyBuilding
   * @return
   */
  void setData(PartyBuilding partyBuilding, PartyBuilding newPartyBuilding);

  /**
   * 编辑党建人员信息
   *
   * @param dutyId
   * @param districtId
   * @param partyBuilding
   * @return
   */
  PartyBuilding update(Long dutyId, Long districtId, PartyBuilding partyBuilding);

  PartyBuilding updateById(Long id, PartyBuilding partyBuilding);

  /**
   * 验证数据是否符合要求
   *
   * @param partyBuilding
   * @return
   */
  void validateField(PartyBuilding partyBuilding);
}

File: web/projects/lib/src/service/party-building.service.ts
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {HttpClient} from '@angular/common/http';
import {Assert} from '@yunzhi/utils';
import {PartyBuilding} from 'projects/lib/src/entity/partyBuilding';

/**
 * 党建(组织)信息
 */
@Injectable({
  providedIn: 'root'
})
export class PartyBuildingService {

  protected baseUrl = 'partyBuilding';

  constructor(protected httpClient: HttpClient,
  ) {
  }

  /**
   * 当前登录用户所在区域的所有党建记录
   */
  getAllOfCurrentUserDistrict(): Observable<PartyBuilding[]> {
    return this.httpClient.get<PartyBuilding[]>(this.baseUrl + '/getAllOfCurrentUserDistrict');
  }

  /**
   * 职务人员新增
   */
  public save(partBuilding: PartyBuilding): Observable<PartyBuilding> {
    // 向后台请求,并通过管道返回User对象
    return this.httpClient.post<PartyBuilding>(`${this.baseUrl}`, partBuilding);
  }

  /**
   * 根据districtId获取该区域内党建人员信息
   */
  public getAllByDistrictId(districtId: number): Observable<Array<PartyBuilding>> {
    Assert.isInteger(districtId, '传入的类型必须是数字');
    return this.httpClient.get<Array<PartyBuilding>>(`${this.baseUrl}/getAllByDistrictId/${districtId.toString()}`);
  }

  public getByDutyIdOfCurrentDistrict(dutyId: number): Observable<PartyBuilding> {
    return this.httpClient.get<PartyBuilding>(this.baseUrl + '/getByDutyIdOfCurrentDistrict/' + dutyId);
  }

  saveOfCurrentDistrict(partBuilding: PartyBuilding): Observable<void> {
    return this.httpClient.post<void>(this.baseUrl + '/saveOfCurrentDistrict', partBuilding);
  }

  getById(id: number): Observable<PartyBuilding> {
    return this.httpClient.get<PartyBuilding>(this.baseUrl + '/' + id);
  }

  update(id: number, partBuilding: PartyBuilding): Observable<void> {
    return this.httpClient.patch<void>(this.baseUrl + '/' + id, partBuilding);
  }

  delete(id: number): Observable<void> {
    return this.httpClient.delete<void>(this.baseUrl + '/' + id);
  }
}
